.script/dataConnectorValidator.ts

```diff
@@ -0,0 +1,38 @@
+import fs from "fs";
+import { runCheckOverChangedFiles } from "./utils/changedFilesValidator";
+import { ExitCode } from "./utils/exitCode";
+import { isValidSchema } from "./utils/jsonSchemaChecker";
+import { isValidId } from "./utils/dataConnectorCheckers/idChecker";
+import * as logger from "./utils/logger";
+
+export async function IsValidDataConnectorSchema(filePath: string): Promise<ExitCode> {
+  let dataConnector = JSON.parse(fs.readFileSync(filePath, "utf8"));
+  let schema = JSON.parse(fs.readFileSync(".script/utils/schemas/DataConnectorSchema.json", "utf8"));
+if(typeof dataConnector.id != "undefined" && typeof dataConnector.connectivityCriterias != "undefined")
+{
+  isValidSchema(dataConnector, schema);
+  isValidId(dataConnector.id);
+}
+else{
+  console.log(`Skipping File path: ${filePath}`);
+}
+
+  return ExitCode.SUCCESS;
+} 
+
+let fileTypeSuffixes = ["*.json"];
+let filePathFolderPrefixes = ["DataConnectors"];
+let fileKinds = ["Added", "Modified"];
+let CheckOptions = {
+  onCheckFile: (filePath: string) => {
+    return IsValidDataConnectorSchema(filePath);
+  },
+  onExecError: async (e: any, filePath: string) => {
+    console.log(`Data Connector Validation Failed. File path: ${filePath}. Error message: ${e.message}`);
+  },
+  onFinalFailed: async () => {
+    logger.logError("An error occurred, please open an issue");
```

Comment:
FYI there is a bug where this message is thrown for every validation error (all of them).
see comment - https://github.com/Azure/Azure-Sentinel/pull/946#issuecomment-675134655
however the fix should be a separate work item and PR